planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -47,74 +47,56 @@ public Itinerary generateItinerary(List<Event> events) {
         int END = events.get(0).getOpeningHours().end();
 
         Collections.sort(events, Event.SortByOrder);
-        Duration [][] travelTimeGraph = getEventsDistanceMatrix(events);
-        if (errorMessage != null) { 
-            Itinerary itinerary = new Itinerary(Arrays.asList(), errorMessage);
-            return itinerary;
+        List<ItineraryItem> items = new ArrayList();
+        if (optimized) {
+            items = scheduleOptimizedItinerary(events, START, END);
+        }else {
+            items = scheduleItineraryInOrder(events, START, END);
         }
-        List<ItineraryItem> items = scheduleItineraryInOrder(events, START, END, travelTimeGraph);
-        Itinerary itinerary = new Itinerary(items, errorMessage);
-        return itinerary;
+        
+        return items;
     }
 
-    // Function that creates a graph with the events as vertices and the travelling
-    // time between the events as edges.
-    // The order of this graph will be the same as the order of the events.
-    // For the MVP, real time traffic is NOT used.
-    // TODO: handle invalid addresses
-    private Duration[][] getEventsDistanceMatrix(List<Event> events) {
-        int numOfNodes = events.size();
-        Duration[][] travelTimeGraph = new Duration[numOfNodes][numOfNodes];
-
-        String GoogleApiKey = "AIzaSyDK36gDoYgOj4AlbCqh1IuaUuTlcpKF0ns";
-        String[] origins = getListOfAddresses(events);
-        String[] destinations = getListOfAddresses(events);
-        GeoApiContext context = new GeoApiContext.Builder(new GaeRequestHandler.Builder())
-                            .apiKey(GoogleApiKey)
-                            .build();
-       try {
-            DistanceMatrix distanceMatrix = DistanceMatrixApi.getDistanceMatrix(context,
-                                                         origins,
-                                                         destinations).await();
+    // Function that creates an itinerary by scheduling each event in order.
+    private List<ItineraryItem> scheduleItineraryInOrder(List<Event> events, int openingTime, int endingTime) 
+                                    throws ItineraryException{
+        List<ItineraryItem> items = new ArrayList();
+        DirectionsRoute directionsRoute = getDirectionsRoute(events, ItineraryOrder.UNOPTIMIZED);
+        DirectionsLeg[] directionsLegs = directionsRoute.legs;
+        int start = openingTime; 
```

Comment:
Check the lines 88 -> 91 of the same file. It's exactly the same piece of code (with a difference in one argument's value).